sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -347,23 +347,23 @@ case class MergeIntoTable(
 }
 
 sealed abstract class MergeAction(
-    condition: Option[Expression]) extends Expression with Unevaluable {
+    val condition: Option[Expression]) extends Expression with Unevaluable {
```

Comment:
to simplify the code a little bit:
```
sealed trait MergeAction extends Expression with Unevaluable {
  def condition: Option[Expression]
  ...
}
```
Then we can just write
```
case class DeleteAction(condition: Option[Expression]) extends MergeAction
```

Reply:
done.